pkgs/tools/filesystems/dwarfs/default.nix

```diff
@@ -0,0 +1,110 @@
+{ stdenv
+, bison
+, boost
+, cmake
+, double-conversion
+, fetchFromGitHub
+, fmt_8
+, fuse3
+, gflags
+, git
+, glog
+, gtest
+, jemalloc
+, lib
+, libarchive
+, libevent
+, libunwind
+, lz4
+, openssl
+, pkg-config
+, ronn
+, xxHash
+, xz
+, zstd
+}:
+
+stdenv.mkDerivation rec {
+  pname = "dwarfs";
+  version = "0.6.1";
+  src = fetchFromGitHub {
+    owner = "mhx";
+    repo = "dwarfs";
+    rev = "v${version}";
+    fetchSubmodules = true;
+    sha256 = "sha256-bGJkgcq8JxueRTX08QpJv1A0O5wXbiIgUY7BrY0Ln/M=";
+  };
+
+  # Needs tagged git repo for generating some files
```

Comment:
Which ones? We should rather monkey patch that instead of creating a git repo and removing metadata information again.

Reply:
See https://github.com/mhx/dwarfs/blob/2a2d089cea3a711e2e7854adcb6dda702c3ba5fd/cmake/version.cmake, these could be manually generated pretty easily. Though I'm not sure if it's worth the effort here? This is a way that won't break via upstream changes like changing the path etc.

Reply:
I feel like we should be taking the path of least resistance (creating a git repository) instead of trying to monkeypatch everything, as it will all just end up being a mess. I'll try getting a monkeypatch functional just for comparison.

Reply:
> I feel like we should be taking the path of least resistance (creating a git repository) instead of trying to monkeypatch everything, as it will all just end up being a mess.

Yeah, I do agree with this (if you see the earlier revisions of my reply). But then I realized you also edit the autogenerated file after, and that can break just as easily as generating it manually.

Reply:
> path of least resistance

That would be to remove https://github.com/mhx/dwarfs/blob/main/cmake/version.cmake#L22-L30 and set the variables via ``-D <var>=<value>``. That won't be a mess, so please go that route.

> I don't think making a Git repo is too far fetched (in fact, [we do it in some other packages](https://cs.github.com/NixOS/nixpkgs?q=%22git+init%22)).

That is a very weak argument in nix because code is not always everywhere up to date.

Reply:
> That would be to remove https://github.com/mhx/dwarfs/blob/main/cmake/version.cmake#L22-L30 and set the variables via ``-D <var>=<value>``. That won't be a mess, so please go that route.

There's also [these lines](https://github.com/mhx/dwarfs/blob/2a2d089cea3a711e2e7854adcb6dda702c3ba5fd/cmake/version.cmake#L44-51) that would need to be removed.

Reply:
Not 100% sure what the definition of "monkeypatch" is here but [my branch](https://github.com/tpwrules/nixpkgs/blob/1fbed16754ff6a1633cfab69e80ffcdcff7274cd/pkgs/tools/filesystems/dwarfs/default.nix#L43) just creates the version files out of whole cloth. Is this a good approach? I personally do disagree with invoking `git` to create a repo.

Reply:
Your approach looks good, seems to work because of how CMake's `execute_process` just doesn't throw an error unconditionally since Git doesn't exist.

Do you mind if it's pulled into this PR?

Reply:
Not at all, please do so.

Reply:
Alright, I pulled in tpwrules's way of patching it, removed the git dependency. In hindsight I should've probably done that (and maybe you should've nagged me to prefer that to the git repository method).

Reply:
> and maybe you should've nagged me to prefer that to the git repository method

Not sure if that was intended to be a dig or not (if so: ouch, but fair), but yeah, I agree. Again, sorry about that.

Reply:
I would recommend to use cmakeFlags instead which should be more stable and achieve the same result. Also instead of manually splitting we should use lib.versions.major etc.

Reply:
> I would recommend to use cmakeFlags instead which should be more stable and achieve the same result.

Disagreed, it would require more patching to `version.cmake` that can break just as easily as this.

> Also instead of manually splitting we should use lib.versions.major etc.

Agreed.

Reply:
> > and maybe you should've nagged me to prefer that to the git repository method
> 
> Not sure if that was intended to be a dig or not (if so: ouch, but fair), but yeah, I agree. Again, sorry about that.

No, no, didn't mean it as a dig. More so as just "tell my stubborn ass to do the better thing"



> Also instead of manually splitting we should use lib.versions.major etc.

I'll do that in the morning.